OHM-Trade-Agent-v1/app/opip/decision/summary.py

```
@@ -239,3 +239,264 @@ def render_scan_summary_text(summary: Mapping[str, Any]) -> str:
         f"Funnel invariant holds: {'YES' if summary.get('invariant_holds') else 'NO'}"
     )
     return "\n".join(lines)
+
+
+def _parse_utc_timestamp(value: Any) -> "datetime | None":
```

Ruff (0.16.3)

[error] Undefined name `datetime`

(F821)